pkg/kubelet/cm/dra/checkpoint/checkpoint.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checkpoint
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/checksum"
+	state "k8s.io/kubernetes/pkg/kubelet/cm/dra/state/v1"
+)
+
+const (
+	CheckpointAPIGroup   = "checkpoint.dra.kubelet.k8s.io"
+	CheckpointKind       = "DRACheckpoint"
+	CheckpointAPIVersion = CheckpointAPIGroup + "/" + state.Version
+)
+
+// Checkpoint represents a structure to store DRA checkpoint data
+type Checkpoint struct {
+	// Data is a JSON serialized checkpoint data
+	// See state.CheckpointData for the details
+	Data string
+	// Checksum is a checksum of Data
+	Checksum checksum.Checksum
```

Comment:
One more thing (a bit more than a nit): can we use some checksum algorithm from the Go standard library instead of `checksum.Checksum`?

I know it's based on a standard Kubernetes util package (https://github.com/kubernetes/kubernetes/blob/master/pkg/util/hash/hash.go), but under the hood it depends on third-party code (spew, see https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/dump/dump.go).

I can't help thinking that this is going to break at some point, and it's complete overkill here because a string is sequence of bytes that can be easily hashed without "spewing" it first back into another string.

Reply:
changed to hash/crc32.ChecksumIEEE from standard library, PTAL.